### PR TITLE
JSON verification for API requests

### PR DESCRIPTION
**Pull request by pkcakeout**
(messed up the branch-name)

- Implements jsonschema verification on gcapi-requests to verify that the API did not change.

### CODE REVIEW

File: gcapi/gcapi.py [hunk 1]
```diff
@@ -21,41 +26,80 @@ def is_uuid(s):
...
+            os.path.dirname(os.path.abspath(__file__)), "schemas", filename
+        )
+
+        try:
+            with open(filename, "r") as f:
+                jsn = json.load(f)
+            return jsonschema.Draft7Validator(jsn)
+        except ValueError as e:
+            print("Error parsing '{}': {}".format(e, filename), file=sys.stderr)
+            return None
```
Comment (jmsmkn) [inline gcapi/gcapi.py]: Should this function return `None` here? Wouldn't it be better to raise an error here and let the callers handle it?

File: gcapi/gcapi.py [hunk 2]
```diff
@@ -148,7 +192,9 @@ def _validate_url(self, url):
         if not url.startswith(self._base_url):
             raise RuntimeError("{} does not start with {}".format(url, self._base_url))
 
-    def __call__(self, method="GET", url="", path="", params=None, json=None, extra_headers={}):
+    def __call__(
+        self, method="GET", url="", path="", params=None, json=None, extra_headers={}
```
Comment (jmsmkn) [inline gcapi/gcapi.py]: `extra_headers` is mutable!

File: gcapi/gcapi.py [hunk 3]
```diff
@@ -67,7 +111,8 @@ def send(self, **kwargs):
             method="POST",
             path=self.base_path,
             json=kwargs,
-            extra_headers={"Content-Type": "application/json"})
+            extra_headers={"Content-Type": "application/json"},
```
Comment (jmsmkn) [inline gcapi/gcapi.py]: Shouldn't every request have this?

File: Pipfile [hunk 4]
```diff
@@ -0,0 +1,12 @@
+[[source]]
```
[the remark this answers is not in the document]
Comment (pkcakeout) [reply Pipfile]: Ha! Did not know about the `-e .` trick. Just looked it up - yeah we can remove the `Pipenv` files then!

**Pull request merged**